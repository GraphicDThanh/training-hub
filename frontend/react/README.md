# React Redux Training
[Plan doc](https://docs.google.com/document/d/1lYywN0RIhrwsyFZ8uPZuaN5QM78mYuXFD608vS8YEbM/edit?pli=1#)
## ES6
## React Basic
### Practice 1 - examples React
## React Advanced
### Practice 2
## Redux Basic
## Redux Advanced
## Redux Recipes
