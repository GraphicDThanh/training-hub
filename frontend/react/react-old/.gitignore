node_modules
out
dist
.DS_Store
.git
.docpad.db
vendor/
.vagrant
bower-components/
bower_components/
npm-debug.log
.gitignore~
vagrant/Vagrantfile~
build/
