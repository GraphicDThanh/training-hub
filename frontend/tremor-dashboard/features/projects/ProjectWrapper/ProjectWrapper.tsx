import Image from "next/image";
import { ReactNode } from "react";

const ProjectWrapper = ({ children }: { children: ReactNode }) => {
  return (
    <div className="-mt-[150px] sm:-mt-[100px]">
      <div className="relative overflow-hidden md:min-w-[320px] min-h-[300px] rounded-xl">
        <Image
          alt="All Projects"
          src={"/images/backgrounds/bg-profile.webp"}
          priority
          fill
          className="object-cover"
          sizes="(max-width: 400px) 33vw, 100vw"
        />
        <div className="absolute w-full h-full bg-project-layer rounded-xl" />
      </div>
      {children}
    </div>
  );
};

export default ProjectWrapper;
