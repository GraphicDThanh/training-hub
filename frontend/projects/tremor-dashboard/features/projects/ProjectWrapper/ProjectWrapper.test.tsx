import { render } from "@testing-library/react";

// Components
import ProjectWrapper from "./ProjectWrapper";

describe("ProjectWrapper", () => {
  const renderComponent = () => {
    return render(
      <ProjectWrapper>
        <div>Test Project Wrapper</div>
      </ProjectWrapper>,
    );
  };

  it("match snapshot", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
