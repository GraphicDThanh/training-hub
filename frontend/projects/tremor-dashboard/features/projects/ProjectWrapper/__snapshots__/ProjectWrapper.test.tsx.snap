// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectWrapper match snapshot 1`] = `
<div>
  <div
    class="-mt-[150px] sm:-mt-[100px]"
  >
    <div
      class="relative overflow-hidden md:min-w-[320px] min-h-[300px] rounded-xl"
    >
      <img
        alt="All Projects"
        class="object-cover"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="(max-width: 400px) 33vw, 100vw"
        src="/_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-profile.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-profile.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-profile.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-profile.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-profile.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-profile.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-profile.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-profile.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-profile.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-profile.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-profile.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <div
        class="absolute w-full h-full bg-project-layer rounded-xl"
      />
    </div>
    <div>
      Test Project Wrapper
    </div>
  </div>
</div>
`;
